Xamarin.Essentials/Platform/Platform.ios.tvos.watchos.cs

```diff
@@ -138,5 +146,21 @@ internal static UIWindow GetCurrentWindow(bool throwIfNull = true)
 
         internal static NSOperationQueue GetCurrentQueue() =>
             NSOperationQueue.CurrentQueue ?? new NSOperationQueue();
+
+#if __IOS__
+        internal class UIPresentationControllerDelegate : UIAdaptivePresentationControllerDelegate
+        {
+            public Action DismissHandler { get; set; }
+
+            public override void DidDismiss(UIPresentationController presentationController) =>
+                DismissHandler?.Invoke();
+
+            protected override void Dispose(bool disposing)
+            {
+                DismissHandler?.Invoke();
```

Comment:
Experimentally, I found out that `DidDismiss` method is not always called when closing parent controllers.  And this is probably the only workaround to finish a `TaskCompletionSource`.